internal/ecrconsumer/git.go

```diff
@@ -0,0 +1,122 @@
+package ecrconsumer
+
+import (
+	"context"
+	"encoding/base64"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/google/go-github/github"
+	"golang.org/x/oauth2"
+)
+
+type GitCommands interface {
+	UpdateFile(msg string, branch string, path string, fileContent []byte) (*github.RepositoryContentResponse, error)
+	DownloadFile(branch string, path string) ([]byte, error)
+}
+
+type GitHub struct {
+	Client       *github.Client
```

Comment:
This field should be private, since we want callers to use the methods on `Github` instead of accessing the `*github.Client` directly.